pkg/client/aad_test.go

```diff
@@ -0,0 +1,77 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/aks-tls-bootstrap-client/pkg/client/mocks"
+	base "github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"go.uber.org/mock/gomock"
+)
+
+var _ = Describe("Aad tests", func() {
+	var (
+		mockCtrl        *gomock.Controller
+		AadClient       = NewAadClient(testLogger)
+		aquireTokenMock *mocks.MockTokenAcquirer
+	)
+
+	BeforeEach(func() {
+		mockCtrl = gomock.NewController(GinkgoT())
+		aquireTokenMock = mocks.NewMockTokenAcquirer(mockCtrl)
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+	})
+
+	Context("Test NewAadClient", func() {
+		It("should return a new AadClient", func() {
+			Expect(AadClient).ToNot(BeNil())
+		})
+	})
+
+	Context("Test GetAadToken", func() {
+		When("newTokenAcquirer is expected to be correct", func() {
```

Comment:
nit: test name is a bit ambiguous, maybe `"newTokenAcquirer returns a valid XXXTokenAcquierer"... It("should return a valid token")`